Add support for Android 15

bhaskar966 opened the pull request:
<!--
Any HTML comment will be stripped when the markdown is rendered, so you don't need to delete them.
-->

### Description
<!--
Describe the changes you have made on a high level in the project.
If this PR is related to an issue, reference it here.
-->

I added support for Android 15 and found a small bug in a function named "clearBitmap" so I fixed that.
To make the app useable and buildable I needed to update the AGP version as well as opt-out from edge to edge view.
Regarding edge-to-edge, I  prefer that you or someone who knows about XML layouts adapt the design according to the guidelines of Andordi 15.

### Motivation
<!--
If this solves a bug, provide the steps to reproduce it or reference the issue, if opened.
In the other cases, specify why this change or this new feature is required or why
it can be helpful to the other users.
-->

I wanted my own apps to support Android 15, so updated here too.

jahirfiquitiva commented on library/src/main/kotlin/dev/jahir/kuper/data/models/Component.kt:
```diff
@@ -120,8 +121,10 @@ data class Component(
-            return Bitmap.createBitmap(newBitmap, minX, minY, maxX - minX + 1, maxY - minY + 1)
+            newBitmap?.setPixels(pixels, 0, width, 0, 0, width, height)
+            return newBitmap?.let { Bitmap.createBitmap(it, minX, minY, maxX - minX + 1, maxY - minY + 1) } ?: bitmap.also {
+                Log.e("clearBitmap", "newBitmap is null")
```
please remove this log

bhaskar966 replied:
Removed the log.

jahirfiquitiva commented on app/src/main/kotlin/dev/jahir/kuper/app/MainActivity.kt:
```diff
@@ -4,7 +4,6 @@ import com.github.javiersantos.piracychecker.PiracyChecker
 import dev.jahir.kuper.ui.activities.KuperActivity
 
 class MainActivity : KuperActivity() {
-
```
please revert this change

bhaskar966 replied:
Now, there are no changes in MainActivity. Sorry for the mistake 😅

bhaskar966 committed:
Fix clearBitmap function
-- we can't create a bitmap which is nullable
now if the bitmap is present it will it will clear it otherwise it will return the actual bitmap.

bhaskar966 committed:
Add support for SDK 35 aka Android 15.

-- Due to my limited knowledge about Views or XML layouts, I'm not able to fix the overlapping content with status bar in Android 15.

Rather I have, opted out from the enforcement. For now it is fine but Google will soon make it deprecated, so we need to adapt the layout for Android 15 before it deprecate.

It is because Android 15 by-default enables edge to edge view.

jahirfiquitiva committed:
Merge branch 'master' into master

jahirfiquitiva commented:
thanks for the help @bhaskar966
